ddtrace/tracer/remote_config.go

```diff
@@ -273,13 +285,63 @@ func (t *tracer) dynamicInstrumentationRCUpdate(u remoteconfig.ProductUpdate) ma
 	return applyStatus
 }
 
-// passProbeConfiguration is used as a stable interface to find the configuration in via bpf. Go-DI attaches
-// a bpf program to this function and extracts the raw bytes accordingly.
+func (t *tracer) dynamicInstrumentationSymDBRCUpdate(
+	u remoteconfig.ProductUpdate,
+) map[string]state.ApplyStatus {
+	applyStatus := make(map[string]state.ApplyStatus, len(u))
+	diRCState.Lock()
+	defer diRCState.Unlock()
+	symDBEnabled := false
+	for k, v := range u {
+		if len(v) == 0 {
+			applyStatus[k] = state.ApplyStatus{State: state.ApplyStateAcknowledged}
```

Comment:
I wonder if we should unmarshal the serialized byte array (`v`) to a structure similar to:
```
type symDbEnablement struct {
		UploadSymbols bool `json:"upload_symbols"`
	}
```
 and set `diRCState.symdbExport` accordingly. You assume that if you have anything in the content, then symdb is enabled.
In the RC of probes, we unmarshal it in the [datadog-agent](https://github.com/DataDog/datadog-agent/blob/main/pkg/dyninst/rcjson/rcjson.go#L28) side, so there it make sense to simply check if there is any data. I think here it differs.

Reply:
Fair enough. The difference from my perspective was that in the datadog agent for the probes we have no choice but to unmarshal it: we need the data. For this the design doc says that the entry is constant. I'm going to need to send another PR to adopt https://github.com/DataDog/datadog-agent/pull/39175 :facepalm: